docs/usage.md

```diff
@@ -101,14 +101,14 @@ In this case, there is no `match` key, so only one rule needs to match:
 `rules` is an array of rules that can be used to target specific files with your defined syntax file.  The rules are processed until the first rule matches, so order your rules in a way that makes sense to you.
 
 ### Filename Rule
-Filename rule defines a filename with regex.  This uses the full file path.
+Filename rule defines a filename with regex.  The pattern is matched against the beginning of the full file path (there is an implicit `^`).
```

Comment:
>  I guess if we are going to mention the ^ we should also mention the $, and maybe mention that there is no adverse effect if they are used as many of the examples use them when it makes no difference (I am guilty myself).

It actually does make a difference whether you use `$` in these rules; e.g., `".*\\.php"` will match `/a/b/c/foo.php5`, where `".*\\.php$"` won't. Not sure if that needs to be documented since it's standard regular-expression behaviour, but i guess it couldn't hurt either. I tried to make the explanation really brief because i didn't want to screw with the documentation style you had going, but i can re-word it however. I'll do another commit in a few minutes and you can see if you like that better.

> But I do feel it was named incorrectly. If I dared change it, I would probably be flooded with issues though...

You could just make it accept both `file_name` and `file_path`, and note in the documentation that the former is deprecated. Would you like me to make a PR for that too?

> By the way, I am grateful for you taking time to help with the documentation.

Thanks, i was afraid you'd think i was nit-picking. I have just recently converted from MacVim, so maybe as i get more into the Sublime ecosystem i can work on this kind of thing more.